share/functions/__fish_svn_prompt.fish

```diff
@@ -0,0 +1,169 @@
+# colour of the revision number to display in the prompt
+set -g __fish_svn_prompt_color_revision yellow
+
+# setting the prompt status separator character
+set -g __fish_svn_prompt_char_separator "|"
+
+# ==============================
+
+# SVN status symbols
+# Do not change these! These are the expected characters that are output from `svn status`, they are taken from `svn help status`
```

Comment:
How about either adding them as local variables to the function they are used in or doing a switch/case there? Or make a list of these characters, another of the names and then map them via index:

```fish
set __fish_svn_status_chars A C D I M N R # ...
set __fish_svn_statuses added conflicted deleted ignored modified replaced # ...
echo $__fish_svn_prompt_char_{$__fish_svn_statuses(contains -i -- $status $__fish_svn_status_chars)}
```

(untested, the parser might barf on that)